packages/junto-utils/api/client.ts

```diff
@@ -1,21 +1,37 @@
 import { Ad4mClient } from "@perspect3vism/ad4m";
-import { ApolloClient, ApolloLink, InMemoryCache } from "@apollo/client";
+import {
+  ApolloClient,
+  InMemoryCache,
+} from "@apollo/client";
 import { WebSocketLink } from "@apollo/client/link/ws";
 
-const PORT = parseInt(window.location.search.slice(6))
-
-const apolloClient = new ApolloClient({
-  link: new WebSocketLink({
-    uri: `ws://localhost:${PORT}/graphql`,
-    options: {
-      reconnect: true,
+export function buildAd4mClient() {
+  const port = parseInt(localStorage.getItem('ad4minPort'))
+  const apolloClient = new ApolloClient({
+    link: new WebSocketLink({
+      uri: `ws://localhost:${port}/graphql`,
```

Comment:
Can we put the whole uri in local storage? so that it can support scenarios like remote hosting and mobile proxy.

Reply:
I have added as a temporary solution and this will change when we implement iframe to host perspective-views.